livekit-agents/livekit/agents/worker.py

```diff
@@ -226,75 +261,93 @@ async def _worker_task(self):
                 scheme = parse.scheme
                 if scheme.startswith("http"):
                     scheme = scheme.replace("http", "ws")
+
                 agent_url = f"{scheme}://{parse.netloc}/{parse.path.rstrip('/')}/agent"
 
-                ws = await self._session.ws_connect(agent_url, headers=headers)
+                ws = await self._session.ws_connect(
+                    agent_url, headers=headers, autoping=True
+                )
+
                 retry_count = 0
 
+                # do worker registration
+                req = agent.WorkerMessage()
+                req.register.type = self._opts.worker_type
+                req.register.allowed_permissions.CopyFrom(
+                    models.ParticipantPermission(
```

Comment:
I don't believe `agent` needs to be set.. and we also need to copy the [can_publish_sources](https://github.com/livekit/protocol/blob/main/protobufs/livekit_models.proto#L80) field